internal/component/otelcol/exporter/splunkhec/config/splunkhec.go

```diff
@@ -0,0 +1,304 @@
+	Enabled bool `alloy:"enabled,attr,optional"`
+	// Override metrics names for telemetry.
+	OverrideMetricsNames map[string]string `alloy:"override_metrics_names,attr,optional"`
+	// extra attributes to be added to telemetry data.
+	ExtraAttributes map[string]string `alloy:"extra_attributes,attr,optional"`
+}
+
+func (args *SplunkHecTelemetry) Convert() *splunkhecexporter.HecTelemetry {
+	if args == nil {
+		return nil
+	}
+	return &splunkhecexporter.HecTelemetry{
+		Enabled:              args.Enabled,
+		OverrideMetricsNames: args.OverrideMetricsNames,
+		ExtraAttributes:      args.ExtraAttributes,
+	}
+}
+
+// SplunkHecClientArguments defines the configuration for the Splunk HEC exporter.
+type SplunkHecArguments struct {
+	SplunkHecClientArguments SplunkHecClientArguments     `alloy:"client,block"`
+	QueueSettings            exporterhelper.QueueSettings `alloy:"queue,block,optional"`
+	RetrySettings            configretry.BackOffConfig    `alloy:"retry_on_failure,block,optional"`
+	Splunk                   SplunkConf                   `alloy:"splunk,block"`
+}
+
+func (args *SplunkHecClientArguments) Convert() *confighttp.ClientConfig {
+	if args == nil {
+		return nil
+	}
+	return &confighttp.ClientConfig{
+		Endpoint:            args.Endpoint,
+		ReadBufferSize:      args.ReadBufferSize,
+		WriteBufferSize:     args.WriteBufferSize,
+		Timeout:             args.Timeout,
+		MaxIdleConns:        args.MaxIdleConns,
+		MaxIdleConnsPerHost: args.MaxIdleConnsPerHost,
+		MaxConnsPerHost:     args.MaxConnsPerHost,
+		IdleConnTimeout:     args.IdleConnTimeout,
+		DisableKeepAlives:   args.DisableKeepAlives,
+		TLSSetting: configtls.ClientConfig{
+			InsecureSkipVerify: args.InsecureSkipVerify,
+		},
+	}
+}
+
+func (args *SplunkHecClientArguments) SetToDefault() {
+	args.Timeout = 15 * time.Second
+}
+
+func (args *SplunkHecClientArguments) Validate() error {
+	if args.Endpoint == "" {
+		return errors.New("missing Splunk hec endpoint")
+	}
+	return nil
+}
+
+func (args *SplunkConf) SetToDefault() {
+	args.BatcherConfig = BatcherConfig{
+		Enabled:      false,
+		FlushTimeout: 200 * time.Millisecond,
+		MinSizeItems: 8192,
+		MaxSizeItems: 0,
+	}
+	args.LogDataEnabled = true
+	args.ProfilingDataEnabled = true
+	args.Source = ""
+	args.SourceType = ""
+	args.Index = ""
+	args.DisableCompression = false
+	args.MaxContentLengthLogs = 2097152
+	args.MaxContentLengthMetrics = 2097152
+	args.MaxContentLengthTraces = 2097152
+	args.MaxEventSize = 5242880
+	args.SplunkAppName = "Alloy"
+	args.SplunkAppVersion = ""
+	args.HealthPath = "/services/collector/health"
+	args.HecHealthCheckEnabled = false
+	args.ExportRaw = false
+	args.UseMultiMetricFormat = false
+	args.Heartbeat = SplunkHecHeartbeat{}
+	args.Telemetry = SplunkHecTelemetry{}
+}
+
+func (args *SplunkConf) Validate() error {
+	if args.Token == "" {
+		return errors.New("missing splunk token")
+	}
+	if !(args.LogDataEnabled || args.ProfilingDataEnabled) {
```

Comment:
nit: I find `if !args.LogDataEnabled && !args.ProfilingDataEnabled` easier to read but feel free to keep it like this if you prefer